pandas/core/series.py

```diff
@@ -1585,6 +1585,56 @@ def __repr__(self) -> str:
 
         return result
 
+    def _repr_html_(self):
+        # TODO: Full independent HTML generation in SeriesFormatter, rather
+        # than depending on a limited subset of functionality via to_frame().
+
+        if get_option("display.notebook_repr_html"):
+            max_rows = get_option("display.max_rows")
+            min_rows = get_option("display.min_rows")
+            show_dimensions = get_option("display.show_dimensions")
+
+            formatter = fmt.DataFrameFormatter(
+                self.to_frame(),
+                columns=None,
+                col_space=None,
+                na_rep="NaN",
+                formatters=None,
+                float_format=None,
+                sparsify=None,
+                justify=None,
+                index_names=True,
+                header=False,
+                index=True,
+                bold_rows=True,
+                escape=True,
+                max_rows=max_rows,
+                min_rows=min_rows,
+                show_dimensions=False,  # We do this later for a series.
+                decimal=".",
+                table_id=None,
+                render_links=False,
+            )
+            html = formatter.to_html(notebook=True).split("\n")
+
+            # Find out where the column ends - we will insert footer information here.
+            tbl_end = [
+                rownum for (rownum, row) in enumerate(html) if "</table>" in row
+            ][-1]
+
+            footer = []
+            if self.name is not None:
+                footer.append("Name: <b>{name}</b>".format(name=self.name))
+            if show_dimensions:
+                footer.append("Length: {rows}".format(rows=len(self)))
+            footer.append("dtype: <tt>{dtype}</tt>".format(dtype=self.dtype))
+
+            html.insert(tbl_end + 1, "<p>{footer}</p>".format(footer=", ".join(footer)))
```

Comment:
Would this still insert paragraph tags when no footer is required at all?

Reply:
Yes, but a footer will always be required so there's no need to check `len(footer)` I believe. Just like the string representation, there will always be at least a `dtype` displayed in the footer. The footer is the main thing that distinguishes between a `Series` and a single-column `DataFrame` (the only other difference is the lack of column headers in `Series`).

I've merged in the latest master and all CI checks pass now.